pkg/server/api/enroll.go

```diff
@@ -94,13 +100,43 @@ func clientAuthenticated(r *http.Request) bool {
 	return len(r.TLS.VerifiedChains) > 0
 }
 
-func clientHostnameMatches(hostname string, r *http.Request) bool {
+func clientHostnameMatches(hostname string, r *http.Request) error {
```

Comment:
that's fair! with VerifyHostname being the prime mover and there being no situation where there could be both true & error, I figured it might be a way to simplify similar to the way that VerifyHostname does it, but I'm certainly not married to the idea and have made the change!

Reply:
I agree that the function reads slightly awkward, so my vote is for matching the signature with `(bool, error)`, would save line 171 as well!